.docker/Dockerfile

```diff
@@ -0,0 +1,30 @@
+# start with official ruby docker image as base
+FROM ruby:3.1.2
+
+# set working directory within container
+WORKDIR /usr/src/app
+
+# pull in ruby (jekyll) and python (cite process) package info
+COPY Gemfile Gemfile.lock _cite/requirements.txt ./
+
+# install ruby packages
+RUN VERSION=$(grep -A 1 'BUNDLED WITH' Gemfile.lock | tail -n 1 | xargs); \
+    gem install bundler --version ${VERSION} && \
+    bundle _${VERSION}_ install
+
+# install python
+RUN apt update && apt install -y python3 python3-pip
```

Comment:
As far as I know I am not using any features in my scripts that wouldn't work on any 3.x. Wouldn't this command always install the latest 3.x anyway? I don't see why this would be needed.

Reply:
IMHO it's fine to leave it as `python3`. Most distros (including Debian Bullseye, on which `ruby:3.1.2` is based) are very conservative about upgrading python, so it's effectively pinned by the distro anyway. If you want to be extra-sure, though, you could restrict it to python 3.9 like so:
```suggestion
RUN apt update && apt install -y python3.9 python3-pip
```
(There doesn't appear to be a similar versioned package for `python3-pip`, but since it's just going to use Python it'll be using 3.9 in this case.)